### PR TITLE
[stable/nextcloud] Add support for Redis, headless service

### PR DESCRIPTION
**Pull request by DaemonDude23**
#### What this PR does / why we need it:

+   Bumps **Nextcloud**'s patch version.
+   Bumps **MariaDB**'s patch version (optional dependency).
+   Adds Support for **Redis**.
+   Adds Support for a headless service with a custom `clusterIP`.
+   Clean's up the README's plaintext table. A few sentences on Redis integration.

#### Which issue this PR fixes

N/A

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by chicken231** [comment]
@chrisingenhaag Anything I can do to get this PR moving?

**Comment by chrisingenhaag** [comment]
/assign @chrisingenhaag 

Hi, didn´t get the info that docker image now supports env vars for Redis, that is great. Thanks for the PR, I will take a look.
One first point for the moment. I´m pretty sure Redis will be used in a prod environment but I´m not sure if we need this enabled by default.

**Comment by chicken231** [comment]
You're right. I didn't mean to leave `redis.enabled` as `true`. I've changed it to `false`.

File: stable/nextcloud/templates/secrets.yaml [hunk 1]
```diff
@@ -15,4 +15,3 @@ data:
   {{ else }}
   nextcloud-password: {{ randAlphaNum 10 | b64enc | quote }}
   {{ end }}
-
```
Comment (chrisingenhaag) [inline stable/nextcloud/templates/secrets.yaml:18]: I think this will break the lint stage in e2e-test.

**Review by chrisingenhaag — COMMENTED** [review]
Can you please add a newline. Everything else looks fine. We can finish PR afterwards.

File: stable/nextcloud/templates/secrets.yaml (hunk 1 above)
Comment (DaemonDude23) [reply stable/nextcloud/templates/secrets.yaml:18]: Newline has been added.

**Comment by chicken231** [comment]
@chrisingenhaag Sorry, it needs sign-off again as I resolved conflicts pertaining to the README/chart version.

**Comment by chrisingenhaag** [comment]
Sorry for the flaky test. I´ll remove it with another PR and find another test method later.

**Comment by chrisingenhaag** [comment]
Flaky test is now removed, you should be able to start a rebase on your branch. Additionally it would be nice if you squash your commits to one "[stable/nextcloud] ...."

**Comment by john-dederer** [comment]
Where did this PR go ? The latest nextcloud chart, which is v1.3.0, has the redis features not included. This PR was for version 1.1.0

**Pull request closed without merging**